bundles/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastAudioSink.java

```diff
@@ -63,19 +83,44 @@ public void process(@Nullable AudioStream audioStream) throws UnsupportedAudioFo
                 if (callbackUrl != null) {
                     // we serve it on our own HTTP server
                     String relativeUrl;
-                    if (audioStream instanceof FixedLengthAudioStream) {
-                        relativeUrl = audioHTTPServer.serve((FixedLengthAudioStream) audioStream, 10);
-                    } else {
-                        relativeUrl = audioHTTPServer.serve(audioStream);
+                    try {
+                        StreamServed streamServed = audioHTTPServer.serve(audioStream, 10, true);
+                        relativeUrl = streamServed.url();
+                        // we have to run the delayed task when the server has completely played the stream
+                        streamServed.playEnd().thenRun(() -> this.playbackFinished(audioStream));
+                    } catch (IOException e) {
+                        logger.warn("Chromecast binding was not able to handle the audio stream (cache on disk failed)",
+                                e);
+                        return;
```

Comment:
You should try to close the audio stream in this case.

Reply:
Indeed ! Thanks, done.